projectile.el

```diff
@@ -3496,6 +3512,31 @@ with a prefix ARG."
                                  :show-prompt arg
                                  :prompt-prefix "Run command: ")))
 
+;;;###autoload
+(defun projectile-repeat-last-command (show-prompt)
+  "Run last projectile external command.
+
+External commands are: `projectile-configure-project',
+`projectile-compile-project', `projectile-test-project' and
+`projectile-run-project'.
+
+If the prefix argument SHOW_PROMPT is non nil, the command can be edited."
+  (interactive "P")
+  (let* ((project-root (projectile-project-root))
```

Comment:
You need to wrap the call to `projectile-project-root` in `projectile-ensure-project`, otherwise you can get a nil outside a project.